NDTensors/src/default_storage.jl

```diff
@@ -0,0 +1,39 @@
+## This is a fil which specifies the default storage type provided some set of parameters
+## The parameters are the element type and storage type
+function default_storage_type(
+  storetype::Type{<:TensorStorage}=Dense,
+  eltype::Type=Float64,
+  datatype::Type{<:AbstractArray}=Vector,
+  index=Tuple{Int},
+)
+  #TODO check if indices are block sparse or not
+  type = nothing
+  try
+    type = storetype{eltype,datatype{eltype}}
+  catch e
+    println("Cannot construct storetype", storetype)
+    throw(e)
+  end
+  return type
+end
+
+default_storage_type(eltype::Type{<:Number}) = default_storage_type(Dense, eltype)
+
+function default_storage_type(datatype::Type{<:AbstractArray})
+  return default_storage_type(Dense, Float64, datatype)
+end
+
+## This function needs to be defined for each vector type
+function default_storage_type(datatype::Type{<:Vector{ElT}}) where {ElT}
```

Comment:
This doesn't seem necessary if the `AbstractArray` is written in a generic way.

Reply:
Yeah I have a more generic method, this was a result of the original code taking the `abstractarray` type without an element type. Once you add element type data its really difficult to separate array type and array element type information